specification/web/resource-manager/Microsoft.Web/2016-03-01/Diagnostics.json

```diff
@@ -0,0 +1,1532 @@
+          "message": {
+            "description": "Message describing the event",
+            "type": "string"
+          },
+          "source": {
+            "description": "Represents the name of the Detector",
+            "type": "string"
+          },
+          "priority": {
+            "format": "double",
+            "description": "Represents the rank of the Detector",
+            "type": "number"
+          },
+          "metaData": {
+            "description": "Downtime metadata",
+            "type": "array",
+            "items": {
+              "type": "array",
+              "items": {
+                "$ref": "./CommonDefinitions.json#/definitions/NameValuePair"
+              }
+            }
+          },
+          "type": {
+            "description": "Represents the type of the Detector",
+            "enum": [
+              "ServiceIncident",
+              "AppDeployment",
+              "AppCrash",
+              "RuntimeIssueDetected",
+              "AseDeployment",
+              "UserIssue",
+              "PlatformIssue",
+              "Other"
+            ],
+            "type": "string",
+            "x-ms-enum": {
+              "name": "IssueType",
+              "modelAsString": false
+            }
+          },
+          "solutions": {
+            "description": "List of proposed solutions",
+            "type": "array",
+            "items": {
+              "$ref": "#/definitions/Solution"
+            }
+          }
+        }
+      },
+      "DetectorDefinition": {
+        "description": "Class representing detector definition",
+        "type": "object",
+        "allOf": [
+          {
+            "$ref": "./CommonDefinitions.json#/definitions/ProxyOnlyResource"
+          }
+        ],
+        "properties": {
+          "properties": {
+            "description": "DetectorDefinition resource specific properties",
+            "properties": {
+              "identifier": {
+                "description": "Name of the detector",
+                "type": "string",
+                "readOnly": true
+              },
+              "displayName": {
+                "description": "Display name of the detector",
+                "type": "string",
+                "readOnly": true
+              },
+              "description": {
+                "description": "Description of the detector",
+                "type": "string",
+                "readOnly": true
+              },
+              "rank": {
+                "format": "double",
+                "description": "Detector Rank",
+                "type": "number",
+                "readOnly": true
+              },
+              "isEnabled": {
+                "description": "Flag representing whether detector is enabled or not.",
+                "type": "boolean",
+                "readOnly": true
+              }
+            },
+            "x-ms-client-flatten": true
+          }
+        }
+      },
+      "DiagnosticAnalysis": {
```

Comment:
id, name and type properties missing for this resource. This should also be a proxy resource and do an allOf on ProxyOnlyresource